course/auth.py

```diff
@@ -82,14 +83,15 @@ def get_pre_impersonation_user(request):
     return None
 
 
-def get_impersonable_user_qset(impersonator):
+def get_impersonable_user_qset(impersonator, course_identifier):
     # type: (User) -> query.QuerySet
-    if impersonator.is_superuser:
-        return User.objects.exclude(pk=impersonator.pk)
+
+    course = Course.objects.get(identifier=course_identifier)
 
     my_participations = Participation.objects.filter(
         user=impersonator,
-        status=participation_status.active)
+        status=participation_status.active,
+        course=course)
```

Comment:
This should only consider the participation in the course to which the impersonation is scoped.